geoextent/__main__.py

```diff
@@ -0,0 +1,92 @@
+
+import sys, os, getopt, datetime
+import geoextent.lib.helpfunctions as hf
+import geoextent.lib.extent as extent
+import logging
+
+COMMAND = None
+
+# the capabilities of our CLI
+def usage():
+    """Provide usage instructions"""
+    return '''
+        NAME
+            geoextent.py 
+
+        Usage example
+            extract_metadata.py -b </absoulte/path/to/directory>
+
+        Supported formats:
+            - (.geojson)
+            - (.csv)
+            - (.shp)
+            - (.geotiff)
+
+        Available options:
+            -b    Extract bounding box
+'''
+
+
+def errorFunction():
+    print("Error: A tag is required for a command")
+    print(usage())
+
+
+def getOutput(filePath, typeOfData):
+    output = extent.fromFile(filePath, typeOfData)
+    if output is None:
+        raise Exception("This file format is not supported")
+    return output
+
+
+if len(sys.argv) == 1:
+    print(usage())
+    sys.exit(1)
+
+
+try:
+    #opts contains tag and path, args contains other args
+    OPTS, ARGS = getopt.getopt(sys.argv[1:], 'b:h')
```

Comment:
Let's switch to `argparse` already now, because it will help us to effectively generate help statements, see  https://docs.python.org/3/library/argparse.html

And example usage in https://github.com/jupyter/repo2docker/blob/master/repo2docker/__main__.py

Reply:
Done, now the library uses argparse to process arguments.